django-workload/django_workload/bundle_tray.py

```diff
@@ -0,0 +1,119 @@
+from .models import (
+    BundleEntryModel,
+    FeedEntryModel,
+    UserModel,
+)
+
+
+INC_FACTOR = 1
+
+
+class BundleTray(object):
+    def __init__(self, request):
+        self.request = request
+
+    def get_bundle(self):
+        bundles = list(
+            BundleEntryModel.objects
+            .filter(userid__in=self.request.user.following).limit(10))
+        # only one bundle per user
+        userids = {}
+        feedentryids = []
+        for bundle in bundles:
+            if bundle.userid in userids:
+                continue
+            userids[bundle.userid] = bundle.id
+            feedentryids += bundle.entry_ids
+        first_bundleids = set(userids.values())
+        # Fetch user information
+        userinfo = {}
+        for user in UserModel.objects.filter(id__in=list(userids)):
+            userinfo[user.id] = user.json_data
+        # fetch entry information
+        feedentryinfo = {}
+        for feedentry in FeedEntryModel.objects.filter(id__in=list(feedentryids)):
+            feedentryinfo[feedentry.id] = {
+                'pk': str(feedentry.id),
+                'comment_count': feedentry.comment_count,
+                'published': feedentry.published.timestamp(),
+            }
+
+        result = {'bundle': [
+            {
+                'pk': str(b.id),
+                'comment_count': b.comment_count,
+                'published': b.published.timestamp(),
+                'user': userinfo[b.userid],
+                'items': [
+                    feedentryinfo[f]
+                    for f in bundle.entry_ids if f in feedentryinfo]
+            }
+            for b in bundles if b.id in first_bundleids
+        ]}
```

Comment:
Whoa, these are some gigantic comprehensions 🙀

Reply:
This was there before in the original workload, I just moved it to a separate file.